editor/import/scene_importer_mesh.cpp

```diff
@@ -0,0 +1,318 @@
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "scene_importer_mesh.h"
+#include "scene/resources/surface_tool.h"
+
+void EditorSceneImporterMesh::add_blend_shape(const String &p_name) {
+	ERR_FAIL_COND(surfaces.size() > 0);
+	blend_shapes.push_back(p_name);
+}
+
+int EditorSceneImporterMesh::get_blend_shape_count() const {
+	return blend_shapes.size();
+}
+
+String EditorSceneImporterMesh::get_blend_shape_name(int p_blend_shape) const {
+	ERR_FAIL_INDEX_V(p_blend_shape, blend_shapes.size(), String());
+	return blend_shapes[p_blend_shape];
+}
+
+void EditorSceneImporterMesh::set_blend_shape_mode(Mesh::BlendShapeMode p_blend_shape_mode) {
+	blend_shape_mode = p_blend_shape_mode;
+}
+
+Mesh::BlendShapeMode EditorSceneImporterMesh::get_blend_shape_mode() const {
+	return blend_shape_mode;
+}
+
+void EditorSceneImporterMesh::add_surface(Mesh::PrimitiveType p_primitive, const Array &p_arrays, const Array &p_blend_shapes, const Dictionary &p_lods, const Ref<Material> &p_material, const String &p_name) {
+	ERR_FAIL_COND(p_blend_shapes.size() != blend_shapes.size());
+	ERR_FAIL_COND(p_arrays.size() != Mesh::ARRAY_MAX);
+	Surface s;
+	s.primitive = p_primitive;
+	s.arrays = p_arrays;
+	s.name = p_name;
+
+	for (int i = 0; i < blend_shapes.size(); i++) {
+		Array bsdata = p_blend_shapes[i];
+		ERR_FAIL_COND(bsdata.size() != Mesh::ARRAY_MAX);
+		Surface::BlendShape bs;
+		bs.arrays = bsdata;
+		s.blend_shape_data.push_back(bs);
+	}
+
+	List<Variant> lods;
+	p_lods.get_key_list(&lods);
+	for (List<Variant>::Element *E = lods.front(); E; E = E->next()) {
+		ERR_CONTINUE(!E->get().is_num());
+		Surface::LOD lod;
+		lod.distance = E->get();
+		lod.indices = p_lods[E->get()];
+		ERR_CONTINUE(lod.indices.size() == 0);
+		s.lods.push_back(lod);
+	}
+
+	s.material = p_material;
+
+	surfaces.push_back(s);
+	mesh.unref();
+}
+int EditorSceneImporterMesh::get_surface_count() const {
```

Comment:
Make sure to always have a separation line between method definitions (we don't group them even if it's setter/getter).

Reply:
Separated.